File: frontend/src/lib/api/analytics.ts
import { api } from '@/lib/api'

export interface VideoStatistics {
  duration: number
  duration_formatted: string
  frame_count: number
  frame_rate: number
  resolution: string
  file_size: number
  file_size_formatted: string
  codec: string
  has_audio: boolean
}

export interface TaskStatistics {
  total_tasks: number
  completed_tasks: number
  failed_tasks: number
  pending_tasks: number
  running_tasks: number
  average_duration: number
}

export interface FileStatistics {
  total_files: number
  total_size: number
  total_size_formatted: string
  by_type: Record<string, number>
}

export interface StageStatistics {
  stage: string
  total_tasks: number
  completed_tasks: number
  failed_tasks: number
  success_rate: number
}

export interface ProjectStatistics {
  project_id: string
  task_stats: TaskStatistics
  file_stats: FileStatistics
  video_stats: VideoStatistics | null
  stage_stats: StageStatistics[]
  created_at: string
  updated_at: string
  last_activity: string | null
}

export interface DashboardMetric {
  title: string
  value: string
  change?: string
  trend?: 'up' | 'down' | 'neutral'
}

export interface DashboardResponse {
  project_id: string
  project_name: string
  metrics: DashboardMetric[]
  statistics: ProjectStatistics
  recent_tasks: Array<{
    id: string
    stage: string
    status: string
    generator_type: string
    created_at: string
    completed_at: string | null
    error_message: string | null
  }>
  files_by_type: Record<string, number>
}

export interface ExportReport {
  project_id: string
  project_name: string
  exported_at: string
  version: string
  statistics: ProjectStatistics
  tasks: Array<Record<string, unknown>>
  files: Array<Record<string, unknown>>
  variants: Array<Record<string, unknown>>
}

export const analyticsApi = {
  /**
   * Get project statistics
   */
  getProjectStats: (projectId: string) =>
    api.get<ProjectStatistics>(`/analytics/projects/${projectId}/stats`).then(r => r.data),

  /**
   * Get dashboard data
   */
  getDashboardData: (projectId: string) =>
    api.get<DashboardResponse>(`/analytics/projects/${projectId}/dashboard`).then(r => r.data),

  /**
   * Export project report as JSON
   */
  exportReport: async (projectId: string) => {
    const response = await api.get(`/analytics/projects/${projectId}/report`, {
      responseType: 'blob',
    })
    // Create download link
    const url = window.URL.createObjectURL(new Blob([response.data]))
    const link = document.createElement('a')
    link.href = url
    link.setAttribute('download', `report_${projectId}.json`)
    document.body.appendChild(link)
    link.click()
    link.remove()
    window.URL.revokeObjectURL(url)
    return response.data
  },

  /**
   * Get video statistics for all video files
   */
  getVideoStatistics: (projectId: string) =>
    api.get(`/analytics/projects/${projectId}/videos`).then(r => r.data),

  /**
   * Analyze a video file
   */
  analyzeVideo: (fileId: string) =>
    api.post(`/analytics/files/${fileId}/analyze`).then(r => r.data),
}

File: backend/app/schemas/analytics.py
"""
Analytics schemas for Phase 14: Data Analysis & Reporting
"""
from uuid import UUID
from datetime import datetime
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field, ConfigDict


class VideoStatistics(BaseModel):
    """Video file statistics"""
    duration: float = 0.0  # in seconds
    duration_formatted: str = "00:00"
    frame_count: int = 0
    frame_rate: float = 0.0
    resolution: str = "0x0"
    file_size: int = 0
    file_size_formatted: str = "0 B"
    codec: str = "unknown"
    has_audio: bool = False

    model_config = ConfigDict(from_attributes=True)


class TaskStatistics(BaseModel):
    """Task execution statistics"""
    total_tasks: int = 0
    completed_tasks: int = 0
    failed_tasks: int = 0
    pending_tasks: int = 0
    running_tasks: int = 0
    average_duration: float = 0.0  # in seconds

    model_config = ConfigDict(from_attributes=True)


class StageStatistics(BaseModel):
    """Per-stage statistics"""
    stage: str
    total_tasks: int = 0
    completed_tasks: int = 0
    failed_tasks: int = 0
    success_rate: float = 0.0  # percentage

    model_config = ConfigDict(from_attributes=True)


class FileStatistics(BaseModel):
    """File statistics"""
    total_files: int = 0
    total_size: int = 0  # in bytes
    total_size_formatted: str = "0 B"
    by_type: Dict[str, int] = {}  # file_type -> count

    model_config = ConfigDict(from_attributes=True)


class ProjectStatistics(BaseModel):
    """Project-level statistics"""
    project_id: UUID
    task_stats: TaskStatistics
    file_stats: FileStatistics
    video_stats: Optional[VideoStatistics] = None
    stage_stats: List[StageStatistics] = []
    created_at: datetime
    updated_at: datetime
    last_activity: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class DashboardMetric(BaseModel):
    """Dashboard metric card"""
    title: str
    value: str
    change: Optional[str] = None
    trend: Optional[str] = None  # "up", "down", "neutral"


class DashboardResponse(BaseModel):
    """Dashboard data response"""
    project_id: UUID
    project_name: str
    metrics: List[DashboardMetric]
    statistics: ProjectStatistics
    recent_tasks: List[Dict[str, Any]] = []
    files_by_type: Dict[str, int] = {}

    model_config = ConfigDict(from_attributes=True)


class ExportReport(BaseModel):
    """Export report structure"""
    project_id: UUID
    project_name: str
    exported_at: datetime
    version: str = "1.0.0"
    statistics: ProjectStatistics
    tasks: List[Dict[str, Any]] = []
    files: List[Dict[str, Any]] = []
    variants: List[Dict[str, Any]] = []

    model_config = ConfigDict(from_attributes=True)
